internal/pkg/ai-functions/functions/test-case/function.go

```diff
@@ -89,32 +101,89 @@ func (f *Function) CompletionOptions() []sdk.PatchOption {
 	}
 }
 
-func (f *Function) Callback(ctx context.Context, arguments json.RawMessage) (any, error) {
+func (f *Function) Callback(ctx context.Context, arguments json.RawMessage, input interface{}, needAdjust bool) (any, error) {
+	testCaseInput, ok := input.(TestCaseFunctionInput)
+	if !ok {
+		err := errors.Errorf("input %v with type %T is not valid for AI Function %s", input, input, Name)
+		return nil, errors.Wrap(err, "bad request: invalid input")
+	}
+
+	bdl := bundle.New(bundle.WithErdaServer())
+	// 根据 issueID 获取对应的需求 Title
+	issue, err := bdl.GetIssueByIDAndUser(testCaseInput.IssueID, f.background.UserID)
+	if err != nil {
+		return nil, errors.Wrap(err, "get requirement info failed")
+	}
+	if issue.Type != apistructs.IssueTypeRequirement {
+		return nil, errors.Wrap(err, "bad request: issue is not type REQUIREMENT")
+	}
+
 	var req apistructs.TestCaseCreateRequest
 	if err := json.Unmarshal(arguments, &req); err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "Unmarshal arguments to TestCaseCreateRequest failed")
+	}
+	f.adjustTestCaseCreateRequest(&req, testCaseInput.TestSetID, testCaseInput.Prompt, issue)
+
+	// 需要调整，则返回创建测试用例的请求 apistructs.TestCaseCreateRequest
+	if needAdjust {
+		return TestCaseMeta{
+			Req:             req,
+			RequirementName: issue.Title,
+			RequirementID:   uint64(issue.ID),
+		}, nil
 	}
 
-	// todo: 背景信息如何获取
-	req.Name = f.UserMessage()
-	req.ProjectID = f.background.ProjectID
-	req.Desc = fmt.Sprintf("Powered by AI.\n\n对应需求:\n%s", f.UserMessage())
-	referer := f.background.GetReferer()
-	u, err := url.Parse(referer)
+	// 无需调整，则返回创建测试用例的请求 apistructs.TestCaseCreateRequest 以及创建成功之后对应的 testcaseID
+	responseBodyData, err := bdl.CreateTestCase(req)
```

Comment:
You should adjust bundle.CreateTestCase and return structured result at bundle response.

Reply:
ok